clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractStickyAssignor.java

```diff
@@ -122,6 +131,13 @@ private boolean allSubscriptionsEqual(Set<String> allTopics,
                 // If the current member's generation is higher, all the previously owned partitions are invalid
                 if (memberData.generation.isPresent() && memberData.generation.get() > maxGeneration) {
                     membersWithOldGeneration.addAll(membersOfCurrentHighestGeneration);
+
+                    allPreviousPartitionsToOwner.clear();
+                    partitionsWithMultiplePreviousOwners.clear();
+                    for (String droppedOutConsumer : membersWithOldGeneration) {
+                        consumerToOwnedPartitions.get(droppedOutConsumer).clear();
+                    }
```

Comment:
This part I just moved here to keep things up to date as we go, before we were clearing them after the loop

Reply:
Because we cleared them earlier now, the `membersWithOldGeneration` is not necessary any more. We can just iterate `membersOfCurrentHighestGeneration` here.

Reply:
Mm good catch